### ISSUE TITLE
Terrible results with pretrained word embeddings

### ISSUE THREAD

**Issue by pythonBerg** [fchollet/keras#5185]
Using the git master version as-is. Downloaded both news and glove from specified sources. Using python 3.5.2 on UBUNTU VM on windows host. Keras version 1.2.1., numpy and tensorflow updated..

Unexpected results showing no predictive value. There must be some basic problem with data or modules or I am reading results incorrectly. I am only starting to experiment with Keras so not many of my own ideas...Anything dumb that I've missed? Thanks

Using TensorFlow backend.
Indexing word vectors.
Found 400000 word vectors.
Processing text dataset
Found 19997 texts.
Found 214873 unique tokens.
Shape of data tensor: (19997, 1000)
Shape of label tensor: (19997, 20)
Preparing embedding matrix.
Training model.
Train on 15998 samples, validate on 3999 samples
Epoch 1/2
15998/15998 [==============================] - 568s - loss: 14.9897 - acc: 0.0503 - val_loss: 15.3080 - val_acc: 0.0503
Epoch 2/2
15998/15998 [==============================] - 561s - loss: 15.3131 - acc: 0.0499 - val_loss: 15.3080 - val_acc: 0.0503
The program finished and will be restarted

**Comment by ddofer** [comment]
What dataset are you working on?

**Comment by pythonBerg** [comment]
as per online code...

news group sample from
http://www.cs.cmu.edu/afs/cs.cmu.edu/project/theo-20/www/data/news20.html
(news20.tar.gz)
glove embeddings from nlp.stanford.edu/data/glove6B.zip

I have not deviated from the current code of git except for modifying the
BASE_DIR


BASE_DIR = '/media/sf_Home'
GLOVE_DIR = BASE_DIR + '/glove.6B/'
TEXT_DATA_DIR = BASE_DIR + '/20_newsgroup/'
MAX_SEQUENCE_LENGTH = 1000
MAX_NB_WORDS = 20000
EMBEDDING_DIM = 100
VALIDATION_SPLIT = 0.2

# first, build index mapping words in the embeddings set
# to their embedding vector

print('Indexing word vectors.')

embeddings_index = {}
f = open(os.path.join(GLOVE_DIR, 'glove.6B.100d.txt'))
for line in f:
    values = line.split()
    word = values[0]
    coefs = np.asarray(values[1:], dtype='float32')
    embeddings_index[word] = coefs
f.close()

print('Found %s word vectors.' % len(embeddings_index))

# second, prepare text samples and their labels
print('Processing text dataset')

texts = []  # list of text samples
labels_index = {}  # dictionary mapping label name to numeric id
labels = []  # list of label ids
for name in sorted(os.listdir(TEXT_DATA_DIR)):
    path = os.path.join(TEXT_DATA_DIR, name)
    if os.path.isdir(path):
        label_id = len(labels_index)
        labels_index[name] = label_id
        for fname in sorted(os.listdir(path)):
            if fname.isdigit():
                fpath = os.path.join(path, fname)
                if sys.version_info < (3,):
                    f = open(fpath)
                else:
                    f = open(fpath, encoding='latin-1')
                texts.append(f.read())
                f.close()
                labels.append(label_id)

print('Found %s texts.' % len(texts))

**Comment by vincefavilla** [comment]
I just tried running this example myself and I'm having the exact same problem. Good to see I'm not the only one.

**Comment by laudney** [comment]
I have the same problem. After fixing the UTF-8 encoding problem, the code runs but immediately loss becomes nan. I've switched to a much lower learning rate of 1e-4 (default is 1e-2) and it manages to train about 25 epochs, reaching val_acc of 25% before imploding.

I've also played with different init for the conv layers without luck.

In any case, I cannot reproduce the claim in the example that it reaches 95% val_acc after 2 epochs.

**Comment by sleeplessTLV** [comment]
As far as I can tell this is still an issue. The example does not reproduce the claimed results - far from it.

**Comment by LABrainTeam** [comment]
I just gone through code and something really surprising and i am quite upset.The accuracy is far less than keras dev team claimed 95%.

Now if you look at the word_index key , i mean the words then you will find that allmost all words are quite meaninglessly rubbish and when you work with GLOVe it really fails to perform a good semantic represeantation.

May be i am wrong but it might be the issue that it fails to get the semantic cosine distance in the embedding or vector space.

I think run this algorithm with reuters dataset will show you better result.Check out Reuters dataset which you will find in keras.datasets module.

If anyone find something about it let me know.

**Comment by DanielDIQA** [comment]
Hi there,
the stated "95% val accuracy" are correct....if you use the binary_accuracy, which is too optimistic in this example here (sparse one-hot-encoded vectors).
Probably the author of this example used in an earlier version the 'binary_crossentropy' loss function, which returns the binary_accuracy as "accuracy" during training.
I found this stackoverflow-reply helpfull:
[https://stackoverflow.com/questions/42081257/keras-binary-crossentropy-vs-categorical-crossentropy-performance](url)

Cheers
Daniel
